compiler/patterns.nim

```diff
@@ -129,46 +129,46 @@ proc matchNested(c: PPatternContext, p, n: PNode, rpn: bool): bool =
       result = bindOrCheck(c, p.sons[2].sym, arglist)
 
 proc matches(c: PPatternContext, p, n: PNode): bool =
-  # hidden conversions (?)
-  if nfNoRewrite in n.flags:
+  let nn = skipConv(n) # hidden conversions
```

Comment:
Thinking about it, ``skipConv`` is not what we need here. We need to skip any ``nkHidden*`` trees since these are not visible.